devito/visitors.py

```diff
@@ -143,6 +149,51 @@ def always_rebuild(self, o, *args, **kwargs):
         return o._rebuild(*new_ops, **okwargs)
 
 
+class PrintAST(Visitor):
```

Comment:
The printing of ``cgen`` object should probably only happen in verbose mode (or equivalent).  The most useful thing, I think, is to see the high level tree structure without the "cgen pollution". For example, think about TTI: you can have thousands cgen statements, and easily lose the focus of what's happening if we print *everything*